airflow/utils/log/log_reader.py

```diff
@@ -54,6 +57,8 @@ def read_log_chunks(
         contain information about the task log which can enable you read logs to the
         end.
         """
```

Comment:
What about `ExternalTaskMarker`? It extends DummyOperator. Should we use

Reply:
We probably need something like `ti.has_logs`. This also avoids duplicated check for `ti.operator` which can be difficult to maintain.

Reply:
yes we should

Reply:
added a `has_logs` to this utility is that sufficient? I didn't want to touch `TaskInstance` if this utility would only be used here.

Reply:
What do you think about using ti.task.is_dummy to check if an operator class is inheriting from DummyOperator?
See:
https://github.com/apache/airflow/blob/a54f8d6b0ce9fd2a2387ccbc11f543d86212ca35/airflow/serialization/serialized_objects.py#L484
https://github.com/apache/airflow/blob/a54f8d6b0ce9fd2a2387ccbc11f543d86212ca35/airflow/www/views.py#L1240-L1242

Reply:
@mik-laj that's a great idea! looks like `ti.task` should be set in this context (based on your link to views.py). Do you think it's good to have some fall back as well? do you see a future where other operator types that don't set this attribute would not have logs? 

```python3
def has_logs(ti: TaskInstance) -> bool:
    if hasattr(ti, 'task'):  # task is not set in __init__ so may be unset
        return not getattr(ti.task, "inherits_from_dummy_operator", False)
    return ti.operator not in OPERATORS_WITHOUT_LOGS
```